sklearn/base.py

```diff
@@ -318,14 +318,20 @@ def __getstate__(self):
         except AttributeError:
             state = self.__dict__.copy()
 
-        if type(self).__module__.startswith("sklearn."):
-            return dict(state.items(), _sklearn_version=__version__)
+        if (
+            type(self).__module__.startswith("sklearn.")
+            and "_sklearn_pickle_version" not in state.keys()
+        ):
```

Comment:
I think `_sklearn_pickle_version` should be overridden every time. This way `_sklearn_pickle_version` means: "the runtime version of the model when it was pickled". 


```suggestion
        if  type(self).__module__.startswith("sklearn."):
```

Let's say a model was trained on `0.24.2` and pickled. Then it is loaded into `1.0.2` and we correctly have `_sklearn_pickle_version=0.24.2` and raise a warning. If the model was fitted again on `1.0.2`, the state would be reset and require the `1.0.2` to run. This means `_sklearn_pickle_version` should be updated to `1.0.2` when pickled.

There is an edge case where a model was trained on on `0.24.2`, pickled, loaded on `1.0.2`, **repickled again on 1.0.2** and a user expects  `_sklearn_pickle_version` to be `0.24.2`. I am considering this usage error.